target/linux/ramips/image/mt7621.mk

```diff
@@ -48,6 +48,20 @@ define Build/iodata-factory
 	fi
 endef
 
+define Build/iodata-mstc-header
+  ( \
+    data_size_crc="$$(dd if=$@ ibs=64 skip=1 2> /dev/null | gzip -c | tail -c 8 | od -An -tx8 --endian little | tr -d ' \n')"; \
+    echo -ne "$$(echo $$data_size_crc | fold -s2 | xargs -I {} echo \\x{} | tr -d '\n')" | \
+      dd of=$@ bs=8 count=1 seek=7 conv=notrunc 2> /dev/null; \
+  )
+  dd if=/dev/zero of=$@ bs=4 count=1 seek=1 conv=notrunc 2> /dev/null
+  ( \
+    header_crc="$$(dd if=$@ bs=64 count=1 2> /dev/null | gzip -c | tail -c 8 | od -An -N4 -tx4 --endian little | tr -d ' \n')"; \
+    echo -ne "$$(echo $$header_crc | fold -s2 | xargs -I {} echo \\x{} | tr -d '\n')" | \
```

Comment:
I'd suggest `echo -ne "$$(echo $$header_crc | sed 's/../\\x&/g')" | \`.

Reply:
It seems to better, I'll try.